handler/src/main/java/io/netty/handler/ssl/JdkSslClientContext.java

```diff
@@ -244,23 +254,28 @@ public JdkSslClientContext(File trustCertCollectionFile, TrustManagerFactory tru
             File keyCertChainFile, File keyFile, String keyPassword, KeyManagerFactory keyManagerFactory,
             Iterable<String> ciphers, CipherSuiteFilter cipherFilter, JdkApplicationProtocolNegotiator apn,
             long sessionCacheSize, long sessionTimeout) throws SSLException {
-        super(newSSLContext(toX509CertificatesInternal(
+        super(newSSLContext(null, toX509CertificatesInternal(
                 trustCertCollectionFile), trustManagerFactory,
                 toX509CertificatesInternal(keyCertChainFile), toPrivateKeyInternal(keyFile, keyPassword),
                 keyPassword, keyManagerFactory, sessionCacheSize, sessionTimeout), true,
                 ciphers, cipherFilter, apn, ClientAuth.NONE, false);
     }
 
-    JdkSslClientContext(X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
-                        X509Certificate[] keyCertChain, PrivateKey key, String keyPassword,
-                        KeyManagerFactory keyManagerFactory, Iterable<String> ciphers, CipherSuiteFilter cipherFilter,
-                        ApplicationProtocolConfig apn, long sessionCacheSize, long sessionTimeout) throws SSLException {
-        super(newSSLContext(trustCertCollection, trustManagerFactory, keyCertChain, key, keyPassword,
-                keyManagerFactory, sessionCacheSize, sessionTimeout), true,
-                ciphers, cipherFilter, toNegotiator(apn, false), ClientAuth.NONE, false);
+    JdkSslClientContext(Provider sslContextProvider,
+        X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
+        X509Certificate[] keyCertChain, PrivateKey key, String keyPassword,
+        KeyManagerFactory keyManagerFactory, Iterable<String> ciphers,
+        CipherSuiteFilter cipherFilter,
+        ApplicationProtocolConfig apn, long sessionCacheSize, long sessionTimeout)
+        throws SSLException {
+        super(newSSLContext(sslContextProvider, trustCertCollection, trustManagerFactory,
+            keyCertChain, key, keyPassword,
+            keyManagerFactory, sessionCacheSize, sessionTimeout), true,
+            ciphers, cipherFilter, toNegotiator(apn, false), ClientAuth.NONE, false);
     }
 
-    private static SSLContext newSSLContext(X509Certificate[] trustCertCollection,
+    private static SSLContext newSSLContext(Provider sslContextProvider,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00107)